clients/src/main/java/org/apache/kafka/clients/admin/MemberAssignment.java

```diff
@@ -19,21 +19,24 @@
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.utils.Utils;
 
-import java.util.List;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
 
 /**
  * A description of the assignments of a specific group member.
  */
 public class MemberAssignment {
-    private final List<TopicPartition> topicPartitions;
+    private final Set<TopicPartition> topicPartitions;
 
     /**
      * Creates an instance with the specified parameters.
      *
      * @param topicPartitions List of topic partitions
      */
-    public MemberAssignment(List<TopicPartition> topicPartitions) {
-        this.topicPartitions = topicPartitions;
+    MemberAssignment(Set<TopicPartition> topicPartitions) {
+        this.topicPartitions = topicPartitions == null ? Collections.<TopicPartition>emptySet() :
+            Collections.unmodifiableSet(new TreeSet<TopicPartition>(topicPartitions));
```

Comment:
nits: `TopicPartition` in `<>` is probably not needed.

Also, do we need any specific feature from TreeSet<> here? If we don't need order of elements then should we use `HashSet` to make the code a bit more consistent and efficient?

Reply:
I used TreeSet here to give a consistent order of listing.

Reply:
Cool. Sounds good.